src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.EasyRequest.cs

```diff
@@ -542,9 +542,10 @@ private void SetProxyCredentials(NetworkCredential credentials)
 
                     // Unlike normal credentials, proxy credentials are URL decoded by libcurl, so we URL encode 
                     // them in order to allow, for example, a colon in the username.
+                    string userNameColonPassword = WebUtility.UrlEncode(credentials.UserName) + ":" + WebUtility.UrlEncode(credentials.Password);
                     string credentialText = string.IsNullOrEmpty(credentials.Domain) ?
-                        string.Format("{0}:{1}", WebUtility.UrlEncode(credentials.UserName), WebUtility.UrlEncode(credentials.Password)) :
-                        string.Format("{2}\\{0}:{1}", WebUtility.UrlEncode(credentials.UserName), WebUtility.UrlEncode(credentials.Password), WebUtility.UrlEncode(credentials.Domain));
+                        userNameColonPassword :
+                        WebUtility.UrlEncode(credentials.Domain) + "\\" + userNameColonPassword;
```

Comment:
An improvement in what measurement?  Seems like for the else condition there are now two string allocations (one for combining the username with the password, and then one for combining that with the domain) where there used to only be one (from the String.Format).